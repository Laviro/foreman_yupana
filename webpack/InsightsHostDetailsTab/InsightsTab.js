import React from 'react';
import PropTypes from 'prop-types';
import { orderBy } from 'lodash';
import { Grid, ListView, noop } from 'patternfly-react';
import ListItem from './components/ListItem';
import './InsightsTab.scss';

class InsightsHostDetailsTab extends React.Component {
  componentDidMount() {
    const { fetchHits, hostID } = this.props;
    fetchHits(hostID);

    const { hash } = window.location;
    if (hash === '#insights') {
      document.querySelector(`li > a[href='${hash}']`).click();
    }
  }

  render() {
    const { hits } = this.props;

    if (!hits.length) {
      return <h2>No recommendations were found for this host!</h2>;
    }
    const hitsSorted = orderBy(hits, ['insights_hit.total_risk'], ['desc']);
    const items = hitsSorted.map(
      (
        {
          insights_hit: {
            title,
            total_risk: totalRisk,
            results_url: resultsUrl,
            solution_url: solutionUrl,
          },
        },
        index
      ) => (
        <ListItem
          key={index}
          title={title}
          totalRisk={totalRisk}
          resultsUrl={resultsUrl}
          solutionUrl={solutionUrl}
        />
      )
    );
    return (
      <div id="host_details_insights_tab">
        <Grid.Row>
          <Grid.Col xs={12}>
            <h2>Recommendations</h2>
            <ListView id="hits_list">{items}</ListView>
          </Grid.Col>
        </Grid.Row>
      </div>
    );
  }
}

InsightsHostDetailsTab.propTypes = {
  hostID: PropTypes.number.isRequired,
  fetchHits: PropTypes.func,
  hits: PropTypes.array,
};

InsightsHostDetailsTab.defaultProps = {
  fetchHits: noop,
  hits: [],
};

export default InsightsHostDetailsTab;
