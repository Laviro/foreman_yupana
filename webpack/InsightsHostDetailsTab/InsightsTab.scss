#host_details_insights_tab {
  #btn_toolbar {
    float: right;
  }

  #btn_fullscreen {
    margin-left: 5px;

    .fa-arrows-alt {
      margin-left: 5px;
    }
  }

  h2 {
    margin-top: 5px;
  }

  #hits_list {
    max-height: 650px;
    overflow-y: scroll;
    overflow-x: hidden;
    margin-top: 15px;

    .list-view-pf-expand {
      padding: 0;
    }

    .list-group-item-header {
      .list-view-pf-expand .fa-angle-right {
        margin-top: 6px;
      }

      .list-view-pf-main-info .list-view-pf-description {
        width: 83%;

        .list-group-item-heading {
          width: 375px;
        }

        p {
          margin: 0;
          font-size: 14px;
        }
      }

      .list-view-pf-additional-info {
        .risk-label {
          padding: 5px 8px;
          border-radius: 12px;

          &.risk-1 {
            background-color: #e7f1fa;
          }

          &.risk-2 {
            background-color: #fdf7e7;
          }

          &.risk-3 {
            background-color: #f9dddd;
          }

          &.risk-4 {
            background-color: #ffecec;
          }

          p {
            margin: 0;
          }
        }
      }
    }

    &::-webkit-scrollbar {
      width: 12px;
      height: 12px;
    }

    &::-webkit-scrollbar-thumb {
      background: #222;
      border-radius: 6px;
      border: 3px solid transparent;
      background-clip: content-box;
    }
  }
}

#new_host_details_insights_tab {
  padding: 24px;
}

#rh-cloud-total-risk-card {
  svg {
    tspan:nth-child(2) {
      fill: black !important;
    }

    g:nth-child(2) {
      path {
        cursor: pointer;
      }
    }

    g:nth-child(3) {
      a {
        text {
          tspan {
            fill: var(--pf-global--link--Color) !important;
            text-decoration: underline;
            text-decoration-color: var(--pf-global--link--Color) !important;
          }
        }
      }
    }
  }
}
