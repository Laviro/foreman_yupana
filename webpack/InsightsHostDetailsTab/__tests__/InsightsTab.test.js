import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import InsightsTab from '../InsightsTab';
import { props } from './InsightsTab.fixtures';

const fixtures = {
  'render with props': props,
};

describe('InsightsTab', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(InsightsTab, fixtures));
});
