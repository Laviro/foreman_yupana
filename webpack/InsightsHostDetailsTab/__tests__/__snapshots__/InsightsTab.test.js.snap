// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsTab rendering render with props 1`] = `
<div
  id="host_details_insights_tab"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <h2>
        Recommendations
      </h2>
      <ListView
        className=""
        id="hits_list"
      >
        <ListItem
          key="0"
          resultsUrl="https://cloud.redhat.com/insights/advisor/recommendations/ansible_deprecated_repo%7CANSIBLE_DEPRECATED_REPO/4739b323-a343-4e89-b71b-81991b8dc656/"
          solutionUrl="https://access.redhat.com/node/3359651"
          title="New Ansible Engine packages are inaccessible when dedicated Ansible repo is not enabled"
          totalRisk={2}
        />
      </ListView>
    </Col>
  </Row>
</div>
`;
